src/main/java/com/google/sps/utility/GeocodingResultUtility.java

```diff
@@ -18,57 +18,79 @@
 import com.google.maps.model.GeocodingResult;
 import com.google.maps.model.LatLng;
 import com.google.maps.model.PlaceType;
+import java.util.Arrays;
+import java.util.List;
 import java.util.Optional;
 
 /** Utility class to extract data from GeocodingResult objects. */
 public class GeocodingResultUtility {
   /**
-   * Parses for coordinates in a resulting call to the Geocoding API.
+   * Parses for the first coordinate found in a resulting call to the Geocoding API.
    *
    * @param result A GeocodingResult returned from the Geocoding API.
    * @return A LatLng representing coordinates.
    */
-  public static LatLng getCoordinates(GeocodingResult result) {
-    return result.geometry.location;
+  public static LatLng getCoordinates(List<GeocodingResult> results) {
+    for (GeocodingResult result : results) {
+      List<AddressType> types = Arrays.asList(result.types);
+      for (AddressType type : types) {
+        if (type == AddressType.STREET_ADDRESS) {
+          return result.geometry.location;
+        }
+      }
+    }
+    return results.get(0).geometry.location;
   }
 
   /**
-   * Converts an AddressType to a PlaceType for the purpose of calling the Places API. PlaceType is
-   * a subset of AddressType and hence, not all AddressTypes are supported by the Places API.
+   * Converts a location to a PlaceType if it exists in the PlaceType enum class.
    *
-   * @param addressType An AddressType to convert to a PlaceType
-   * @return A PlaceType corresponding to an AddressType if available, null if not available.
+   * @param location A string representing a location.
+   * @return An optional containing a PlaceType if an equivalent is found for the location
+   *     specified, an empty optional otherwise.
    */
-  public static Optional<PlaceType> convertAddressTypeToPlaceType(AddressType addressType) {
+  public static Optional<PlaceType> convertToPlaceType(String location) {
     for (PlaceType placeType : PlaceType.values()) {
-      if (placeType.name().equals(addressType.name())) {
+      if ((placeType.name()).equals(location.toUpperCase().replace(" ", "_"))) {
```

Comment:
nit 1: why the extra braces?
nit 2: `equalsIgnoreCase()`

Reply:
nit 1: just thought it was clearer that way but I guess it doesn't make it any clearer
nit 2: oh that's nice, thanks!

Reply:
`RESOLVED`